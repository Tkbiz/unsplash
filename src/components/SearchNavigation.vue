<template>
  <v-container class="search-position pa-xl-0">
    <v-col cols="12"
           class="search-nav">
      <v-row>
        <v-text-field
            v-model="search"
            @change="searchPhotos"
            class="custom-input"
            hide-details
            label="Search for high-resolution photos">
          <v-img
              @click="searchPhotos"
              class="search-icon"
              slot="append"
              src="@/assets/images/icons/search.svg">
          </v-img>
        </v-text-field>
        <div class="trending">
          Trending
        </div>
        <v-chip-group
            active-class="primary--text"
            column>
          <v-chip
              v-for="(tag, index) in tags"
              :key="tag + index"
              @click="searchCollectionPhotos(tag)">
            {{ tag }}
          </v-chip>
        </v-chip-group>
      </v-row>
    </v-col>
  </v-container>
</template>

<script>
import {bus} from '@/bus'

export default {
  name: 'SearchNavigation',
  props: {
    tags: {
      type: Array,
      required: true
    }
  },
  data: () => ({
    search: null,
    page: 1
  }),
  methods: {
    async searchPhotos() {
      let tagName = this.search
      await this.$store.dispatch('searchPhotos', {
        tagName: tagName,
        page: this.page
      })
      bus.$emit('change-photos', this.photosNewList)
      this.$store.commit('fetchTagName', tagName)
      this.search = null
    },
    async searchCollectionPhotos(tag) {
      await this.$store.dispatch('searchPhotos', {
        tagName: tag,
        page: this.page
      })
      bus.$emit('change-photos', this.photosNewList)
      this.$store.commit('fetchTagName', tag)
    }
  },
  computed: {
    photosNewList() {
      return this.$store.getters.photos
    }
  }
}
</script>
