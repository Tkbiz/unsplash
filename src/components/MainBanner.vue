<template>
  <div
      v-if="randomPhoto"
      class="container--fluid main-banner">
    <v-row class="justify-end align-center position-relative ma-0">
      <v-container class="banner-block">
        <v-row>
          <v-col
              class="pa-xl-0"
              cols="12"
              md="8"
              lg="5"
              xl="4">
            <h1 v-if="randomPhoto.user"
                class="banner-title col-12 col-lg-10">
              {{ bannerTitle }}
              <br>
              {{ randomPhoto.user.name | truncate(65, '...') }}
            </h1>
            <p v-if="randomPhoto.description"
               class="banner-description col-12 col-md-7 col-lg-8">
              {{ randomPhoto.description | truncate(130, '...') }}
            </p>
            <v-btn
                class="banner-btn"
                depressed>
              explore all
            </v-btn>
          </v-col>
        </v-row>
      </v-container>
      <v-col
          cols="12"
          lg="7"
          class="position-relative banner-img">
        <v-img
            v-if="randomPhoto.urls"
            :src="randomPhoto.urls.regular ? randomPhoto.urls.regular : '@/assets/images/no-image.png'">
        </v-img>
      </v-col>
      <SearchNavigation :tags="tags"/>
    </v-row>
  </div>
</template>

<script>
import SearchNavigation from '@/components/SearchNavigation'
import {truncateTextMixin} from "@/mixins/truncateTextMixin"

export default {
  name: 'MainBanner',
  mixins: [truncateTextMixin],
  props: {
    tags: {
      type: Array,
      required: true
    }
  },
  data: () => ({
    bannerTitle: 'Photo of the Day by ',
    randomPhoto: {}
  }),
  components: {
    SearchNavigation
  },
  async mounted() {
   this.randomPhoto = await this.$store.dispatch('fetchRandomPhoto')
  }
}
</script>
