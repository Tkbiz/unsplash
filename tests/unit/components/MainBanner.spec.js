import {mount, createLocalVue} from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import MainBanner from '@/components/MainBanner'
import SearchNavigation from '@/components/SearchNavigation'
import {truncateTextMixin} from "@/mixins/truncateTextMixin"
import Vuex from 'vuex'

Vue.use(Vuetify)

const localVue = createLocalVue()
localVue.use(Vuex)

describe('MainBanner.vue', () => {
  let vuetify
  let wrapper
  let actions
  let store

  beforeEach(() => {
    vuetify = new Vuetify()

    actions = {
      fetchRandomPhoto: jest.fn()
    }

    store = new Vuex.Store({
      actions
    })

    wrapper = mount(MainBanner, {
      localVue,
      vuetify,
      store,
      propsData: {
        tags: []
      },
      data: () => ({
        bannerTitle: 'Photo of the Day by ',
        randomPhoto: {}
      })
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it("renders SearchNavigation", () => {
    expect(wrapper.exists(SearchNavigation)).toBe(true)
  })

  it("renders truncateTextMixin", () => {
    expect(wrapper.exists(truncateTextMixin)).toBe(true)
  })

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders props when passed', () => {
    const tags = []

    expect(wrapper.props()).toEqual({tags})
  })

  it('dispatches a fetchRandomPhoto action on mounted hook', () => {
    expect(actions.fetchRandomPhoto).toHaveBeenCalled()
    expect(actions.fetchRandomPhoto).toHaveBeenCalledTimes(1)
    expect(actions.fetchRandomPhoto.mock.calls.length).toBe(1)
  })
})
