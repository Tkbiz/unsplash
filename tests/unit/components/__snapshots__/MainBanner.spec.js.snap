// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainBanner.vue should match snapshot 1`] = `""`;
